spec/abstract/RISCV64/Init_A.thy

```diff
@@ -37,12 +37,20 @@ definition canonical_user :: "vspace_ref" where
 
 (* Kernel and ELF window are constructed so that they can be covered with one max_pt_level entry
    each. This is not the layout the real kernel uses, but we are only trying to show that
-   the invariants are consistent. *)
+   the invariants are consistent.
+
+   The values we pick here for the size of these regions constrain pptr_base and kernel_elf_base
+   in real kernel configurations, so we pick relatively small values that are still large enough
+   to work for the example and that are reasonable lower bounds for real platforms:
+   4M physical memory (1 << 22) and one page (1 << pageBits) for the kernel elf region. *)
+definition kernel_window_bits :: nat where
+  "kernel_window_bits \<equiv> 22"
```

Comment:
This constant could probably be abstracted in a nicer way, but for now I just replaced all the relevant instances of `1 << 20` that I found with `1 << kernel_window_bits` and then unfolded `kernel_window_bits` whenever needed.

Reply:
I think that's good as is. We should probably add a comment that this is just an example size and chosen to be smallish but to fit at least a  `RISCVLargePage` + a minimum set of objects for the InfoFlow example.

Reply:
Ah, I should have looked into the file first :)

Reply:
Most of it is there, but the forward reference to infoflow and the constraints from that would be good to add.